src/features/previewContentProvider.ts

```diff
@@ -86,22 +87,22 @@ export class AsciidocContentProvider {
           data-settings="${JSON.stringify(initialData).replace(/"/g, '&quot;')}"
           data-strings="${JSON.stringify(previewStrings).replace(/"/g, '&quot;')}"
           data-state="${JSON.stringify(state || {}).replace(/"/g, '&quot;')}">
-        <script src="${this.extensionScriptPath('pre.js')}" nonce="${nonce}"></script>
+        <script src="${this.extensionResourcePath(editor.webview, 'pre.js')}" nonce="${nonce}"></script>
         ${this.getStyles(sourceUri, nonce, config, state)}
         <base href="${editor.webview.asWebviewUri(asciidocDocument.uri).toString(true)}">
       </head>
       <body class="${bodyClassesVal} vscode-body ${config.scrollBeyondLastLine ? 'scrollBeyondLastLine' : ''} ${config.wordWrap ? 'wordWrap' : ''} ${config.markEditorSelection ? 'showEditorSelection' : ''}">
         ${body}
         <div class="code-line" data-line="${asciidocDocument.lineCount}"></div>
-        <script async src="${this.extensionScriptPath('index.js')}" nonce="${nonce}" charset="UTF-8"></script>
+        <script async src="${this.extensionResourcePath(editor.webview, 'index.js')}" nonce="${nonce}" charset="UTF-8"></script>
       </body>
       </html>`
   }
 
-  private extensionScriptPath (mediaFile: string): string {
-    return vscode.Uri.file(this.context.asAbsolutePath(path.join('dist', mediaFile)))
-      .with({ scheme: 'vscode-resource' })
-      .toString()
+  private extensionResourcePath (webview: Webview, mediaFile: string): string {
+    const webviewResource = webview.asWebviewUri(
+      vscode.Uri.joinPath(this.context.extensionUri, 'media', mediaFile))
```

Comment:
This code expects `index.js` and `pre.js` to be in the `media` directory, but we do not place them there.
My recollection is that I introduced this distinction between resources and scripts so that we could webpack the preview into a different folder.

If I manually place these files into this folder then the preview scrolling works.

I think we do want these to be in the 'dist' folder somehow because they are built artifacts.

Reply:
(just lost my comment by clicking "unresolve conversation" 😠)

> My recollection is that I introduced this distinction between resources and scripts so that we could webpack the preview into a different folder.

Could you please tell me more about this choice? I don't see any benefit of using a different directory.
For reference, the Markdown extension is also using Webpack with the `media` directory as the `output`. They are also using a `.gitignore` to ignore the generated JavaScript files:

- `webpack.config.js`: https://github.com/microsoft/vscode/blob/66b1668b66768275b655cde14d96203915feca7b/extensions/markdown-language-features/webpack.config.js#L28
- `.gitignore`: https://github.com/microsoft/vscode/blob/66b1668b66768275b655cde14d96203915feca7b/extensions/markdown-language-features/.gitignore#L2

Using the same directory means that "all" webview resources are in the same place. We can then use the same function to resolve the path to `index.js`, `pre.js`, `highlight.css`, `asciidoctor-default.css`...

Reply:
Let's create a follow-up issue.